### PR TITLE
Client and Node Interface changes

### PR DESCRIPTION
**Pull request by inetic**
[Review on Reviewable](https://reviewable.io/reviews/maidsafe/routing/480)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @benjaminbollen

(maidsafe_highfive has picked a reviewer for you, use r? to override)

**Comment by benjaminbollen** [comment]
test this please

**Comment by benjaminbollen** [comment]
can you please merge upstream/master and address the node_interface changes in routing_membrane tests? thanks!

**Comment by inetic** [comment]
done

File: src/node_interface.rs [hunk 1]
```diff
@@ -73,11 +73,11 @@ pub trait Interface : Sync + Send {
...
     /// InterfaceError.
     fn handle_post(&mut self,
-                   our_authority: Authority,
-                   from_authority: Authority,
-                   from_address: NameType,
-                   name : NameType,
-                   data: Data) -> Result<MessageAction, InterfaceError>;
+                   our_authority  : Authority,
+                   from_authority : Authority,
+                   from_address   : NameType,
```
Comment (benjaminbollen) [inline src/node_interface.rs]: we can address on Monday whether from_address is needed, also name is in our_authority and in data itself

**Comment by benjaminbollen** [comment]
looks good to me

**Comment by dirvine** [comment]
Nice one !!

**Pull request merged**